doc/design.md

```diff
@@ -0,0 +1,184 @@
+# Introduction
+
+## Acknowledgments
+
+This document's section structure is derived from the "[arc42](https://arc42.org/)" architectural template by Dr. Gernot Starke, Dr. Peter Hruschka.
```

Comment:
I don't get what's acknowleged here. If it's only the structure of a design, I would at least move it to the end of the file.